### PR TITLE
marine helmets no longer spawn with squad optics. 

### PR DESCRIPTION
**Pull request by private-tristan**
# About the pull request

marine helmets no longer spawn with squad optics. 

can still be found in surplus uniform vendors.

# Explain why it's good for the game

currently squad optics serve no purpose, their only function (showing roles of allies) is already fulfilled by having your HEADSET on. I don't know of any situations where a marine would discard their headset and rely solely on their optics to save them half a second of seeing a medic's helmet to know that they are, indeed, a medic.

removing them saves me from needing to press extra buttons every time i want to swap from NVGs to medhud as an IO.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: Private Tristan
del: marine helmets no longer spawn with squad optics. they have been moved to surplus uniform vendors.
/:cl:

### CODE REVIEW

**Comment by Sulaboy** [comment]
If you’re doing something like this you should make the squad optic available to add to your helmet if you like how it looks. The marines in the movie had the optics.

You could just stick the item in the surplus vendor or flavor menu. It would be even better if you could toggle which huds cycle on the button so you could disable the ones you don’t like.

**Comment by private-tristan** [comment]
> If you’re doing something like this you should make the squad optic available to add to your helmet if you like how it looks. The marines in the movie had the optics.
> 
> You could just stick the item in the surplus vendor or flavor menu. It would be even better if you could toggle which huds cycle on the button so you could disable the ones you don’t like.


sounds good, i'll add them to surplus

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -718,6 +713,7 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
 	icon_state = "tech_helmet"
 	specialty = "M10 technician"
 	built_in_visors = list(new /obj/item/device/helmet_visor, new /obj/item/device/helmet_visor/welding_visor)
+	built_in_visors = list(new /obj/item/device/helmet_visor/welding_visor)
```
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:716]: ?

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -514,11 +514,6 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
 		inserted_visors = list()
 		to_chat(user, SPAN_NOTICE("You remove the inserted visors."))
 
-		var/datum/action/item_action/cycle_helmet_huds/cycle_action = locate() in actions
-		cycle_action.set_default_overlay()
-		if(!length(built_in_visors))
-			cycle_action.remove_from(user)
-
```
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:521]: ?

File: code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm [hunk 3]
```diff
@@ -147,6 +147,7 @@
 		list("Rebreather", round(scale * 10), /obj/item/clothing/mask/rebreather, MARINE_CAN_BUY_MASK, VENDOR_ITEM_REGULAR),
 
 		list("MISCELLANEOUS", -1, null, null, null),
+		list("Squad Optics", round(scale *10), /obj/item/device/helmet_visor, VENDOR_ITEM_REGULAR),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm:150]:
```suggestion
		list("Squad Optics", round(scale * 10), /obj/item/device/helmet_visor, VENDOR_ITEM_REGULAR),
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/helmet.dm (hunk 1 above)
Comment (private-tristan) [reply code/modules/clothing/head/helmet.dm:716]: must have accidentally copied that instead of removing helmet visor, oops.

**Pull request closed without merging**